pkg/query/querier.go

```diff
@@ -159,6 +160,11 @@ type querier struct {
 	selectTimeout           time.Duration
 	shardInfo               *storepb.ShardInfo
 	seriesStatsReporter     seriesStatsReporter
+
+	enablePreAggregationLabelRewrite bool
```

Comment:
it is verbose to introduce so many variable, some of them are actually hard coded like `aggregationLabelName`, you should make them constant.

Also it is better to create a separate file in the package so it is less code injection for merging upstream changes.